pkg/api/norman/customization/app/app.go

```diff
@@ -92,6 +102,33 @@ func (w Wrapper) Validator(request *types.APIContext, schema *types.Schema, data
 	return nil
 }
 
+// ValidateDashboardAppDoesNotExist validates that a request is not attempting
+// to create an apps.project.cattle.io with an ID that matches an
+// apps.catalog.cattle.io in a namespace that matches its targetNamespace.
+func (w Wrapper) ValidateDashboardAppDoesNotExist(clusterName, appID, ns string, request *types.APIContext) error {
+	if request.Method != http.MethodPost {
+		return nil
+	}
+
+	couldNotValidateErrFormat := "could not validate apps.catalog.cattle.io with same ID and namespace does not exist: %v"
+	clusterContext, err := w.ClusterManager.UserContext(clusterName)
+	if err != nil {
+		return httperror.NewAPIError(httperror.ServerError, fmt.Sprintf(couldNotValidateErrFormat, err))
+	}
+
+	appClient := clusterContext.Catalog.V1().App()
+	if _, err := appClient.Get(ns, appID, metav1.GetOptions{}); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return httperror.NewAPIError(httperror.ServerError, fmt.Sprintf(couldNotValidateErrFormat, err))
+		}
+	} else {
+		return httperror.NewAPIError(httperror.Conflict, fmt.Sprintf("an apps.catalog.cattle.io already exists with"+
+			" ID [%s] in namespace [%s] which matches targetNamespace.", appID, ns))
+	}
```

Comment:
nit: no period in errors and I think we don't need the last phrase.

```suggestion
		return httperror.NewAPIError(httperror.Conflict, fmt.Sprintf("an apps.catalog.cattle.io already exists with"+
			" ID [%s] in namespace [%s]", appID, ns))
	}
```